src/main/java/org/cloudfoundry/promregator/scanner/ReactiveTargetResolver.java

```diff
@@ -174,57 +204,26 @@ public void setResolvedApplicationId(String resolvedApplicationId) {
-		
-		Flux<IntermediateTarget> initialFlux = Flux.fromIterable(configTargets)
-				.map(configTarget -> {
-					IntermediateTarget it = new IntermediateTarget();
-					it.setConfigTarget(configTarget);
-					
-					return it;
-				});
-		
-		Flux<IntermediateTarget> orgResolvedFlux = initialFlux.flatMap(this::resolveOrg).log(log.getName()+".resolveOrg");
-		Flux<IntermediateTarget> spaceResolvedFlux = orgResolvedFlux.flatMap(this::resolveSpace).log(log.getName()+".resolveSpace");
-		Flux<IntermediateTarget> applicationResolvedFlux = spaceResolvedFlux.flatMap(this::resolveApplication).log(log.getName()+".resolveApplication")
-				.doOnNext(it -> {
-					if (it.getResolvedOrgId() == null) {
-						log.error(String.format("Target '%s' created a ResolvedTarget without resolved org id", it.getConfigTarget()));
-					}
-					
-					if (it.getResolvedSpaceId() == null) {
-						log.error(String.format("Target '%s' created a ResolvedTarget without resolved space id", it.getConfigTarget()));
-					}
-					
-					if (it.getResolvedApplicationId() == null) {
-						log.error(String.format("Target '%s' created a ResolvedTarget without resolved application id", it.getConfigTarget()));
-					}
-					
-					if (it.getResolvedOrgName() == null) {
-						log.error(String.format("Target '%s' created a ResolvedTarget without resolved org name", it.getConfigTarget()));
-					}
-					
-					if (it.getResolvedSpaceName() == null) {
-						log.error(String.format("Target '%s' created a ResolvedTarget without resolved space name", it.getConfigTarget()));
-					}
-					
-					if (it.getResolvedApplicationName() == null) {
-						log.error(String.format("Target '%s' created a ResolvedTarget without resolved application name", it.getConfigTarget()));
-					}
-				});
-		
-		Flux<ResolvedTarget> resultFlux = applicationResolvedFlux.map(IntermediateTarget::toResolvedTarget);
-		
-		List<ResolvedTarget> resultList = resultFlux.distinct().collectList().block();
-		if (log.isDebugEnabled()) {
-			log.debug(String.format("Successfully resolved %d configuration targets to %d resolved targets", configTargets.size(), resultList.size()));
-		}
-		
-		return resultList;
+		return Flux.fromIterable(configTargets)
+				.parallel()
+				.runOn(Schedulers.parallel())
+				.map(IntermediateTarget::new)
+				.flatMap(intTarget -> resolveOrg(intTarget).parallel().runOn(Schedulers.parallel()))
+				.log(log.getName() + ".resolveOrg")
+				.flatMap(intTarget -> resolveSpace(intTarget).parallel().runOn(Schedulers.parallel()))
+				.log(log.getName() + ".resolveSpace")
+				.flatMap (intTarget -> resolveApplication(intTarget).parallel().runOn(Schedulers.parallel()))
+				.log(log.getName() + ".resolveApplication")
+				.map(IntermediateTarget::toResolvedTarget)
+				.sequential()
+				.distinct().collectList()
+				.doOnNext(it -> log.debug("Successfully resolved {} configuration targets to {} resolved targets", configTargets.size(), it.size()))
+				.block();
```

Comment:
Looking forward to removing this `block`, but will save that work for an upcoming PR.